app/controllers/submission_viewing_events_controller.rb

```diff
@@ -0,0 +1,90 @@
+class SubmissionViewingEventsController < ApplicationController
+  include SubmittedContentHelper
+  def action_allowed?
+    true
+  end
+
+  # record time when link or file is opened in new window
+  def record_start_time
+    param_args = params[:submission_viewing_event]
+    # check if this link is already opened and timed
+    store = LocalStorage.new()
+    submission_viewing_event_records = store.where(map_id: param_args[:map_id], round: param_args[:round], link: param_args[:link])
+    # if opened, end these records with current time
+    if submission_viewing_event_records
+      submission_viewing_event_records.each do |time_record|
+        if time_record.end_at.nil?
+          # time_record.update_attribute('end_at', start_at)
+          store.remove(time_record)
+        end
+      end
+    end
+    # create new response time record for current link
+    # submission_viewing_event = SubmissionViewingEvent.new(submission_viewing_event_params)
+    # submission_viewing_event.save
+    submission_viewing_event = LocalSubmittedContent.new(submission_viewing_event_params)
+    store.save(submission_viewing_event)
+
+    #if creating start time for expertiza update end times for all other links.
+    if param_args[:link]=='Expertiza Review' 
+      params[:submission_viewing_event][:link] = nil
+      params[:submission_viewing_event][:end_at] = params[:submission_viewing_event][:start_at]
+      record_end_time()
+    end
+    render nothing: true
```

Comment:
This is an example of methods from this implementation that I would love to simplify. It doesn't seem like as much code needs to be here as there is, and this is not the only method my sentiment applies to. This is not something I think needs to be done in order to merge this, but I thought I'd share.

Reply:
Totally agree. On my list of things to do